packages/server/lib/controllers/auth/postApiKey.ts

```diff
@@ -72,10 +72,15 @@ export const postPublicApiKeyAuthorization = asyncWrapper<PostPublicApiKeyAuthor
     const queryString: PostPublicApiKeyAuthorization['Querystring'] = queryStringVal.data;
     const { providerConfigKey }: PostPublicApiKeyAuthorization['Params'] = paramsVal.data;
     const connectionConfig = queryString.params ? getConnectionConfig(queryString.params) : {};
-    const connectionId = queryString.connection_id || connectionService.generateConnectionId();
+    let connectionId = queryString.connection_id || connectionService.generateConnectionId();
     const hmac = 'hmac' in queryString ? queryString.hmac : undefined;
     const isConnectSession = res.locals['authType'] === 'connectSession';
 
+    if (isConnectSession && queryString.connection_id) {
+        errorRestrictConnectionId(res);
```

Comment:
at some point we should stop for a minute and reassess all the auth functions and see if we could consolidate some of the shared logic.
we also repeat the reconnect mechanism below so not sure trying to avoid the repetition here and not below is necessary